src/lib/header-bidding/prebid/magnite.ts

```diff
@@ -0,0 +1,190 @@
+	containsWS,
+	getBreakpointKey,
+} from '../utils';
+
+export const getMagniteZoneId = (
+	slotId: string,
+	sizes: HeaderBiddingSize[],
+): number => {
+	if (isInUk()) {
+		switch (getBreakpointKey()) {
+			case 'D':
+				if (containsMpuOrDmpu(sizes) || containsWS(sizes)) {
+					return 3426780;
+				}
+				// top-above-nav on desktop
+				if (
+					containsLeaderboardOrBillboard(sizes) &&
+					slotId === 'dfp-ad--top-above-nav'
+				) {
+					return 3426786;
+				}
+				// Fronts-banners on desktop
+				if (
+					containsBillboard(sizes) &&
+					slotId.includes('fronts-banner')
+				) {
+					return 3426790;
+				}
+				break;
+			case 'M':
+				if (containsMpu(sizes) || containsPortraitInterstitial(sizes)) {
+					return 3426778;
+				}
+				break;
+			default:
+				return -1;
+		}
+	}
+
+	if (isInRow()) {
+		switch (getBreakpointKey()) {
+			case 'D':
+				if (containsMpuOrDmpu(sizes) || containsWS(sizes)) {
+					return 3426822;
+				}
+				// top-above-nav on desktop
+				if (
+					containsLeaderboardOrBillboard(sizes) &&
+					slotId === 'dfp-ad--top-above-nav'
+				) {
+					return 3426828;
+				}
+				// Fronts-banners on desktop
+				if (
+					containsBillboard(sizes) &&
+					slotId.includes('fronts-banner')
+				) {
+					return 3426834;
+				}
+				break;
+			case 'M':
+				if (containsMpu(sizes) || containsPortraitInterstitial(sizes)) {
+					return 3426836;
+				}
+				if (containsMobileSticky(sizes)) {
+					return 3477560;
+				}
+				break;
+			default:
+				return -1;
+		}
+	}
+
+	if (isInUsOrCa()) {
+		switch (getBreakpointKey()) {
+			case 'D':
+				if (containsMpuOrDmpu(sizes) || containsWS(sizes)) {
+					return 3471422;
+				}
+				// top-above-nav on desktop
+				if (
+					containsLeaderboardOrBillboard(sizes) &&
+					slotId === 'dfp-ad--top-above-nav'
+				) {
+					return 3471428;
+				}
+				// Fronts-banners on desktop
+				if (
+					containsBillboard(sizes) &&
+					slotId.includes('fronts-banner')
+				) {
+					return 3471434;
+				}
+				break;
+			case 'M':
+				if (containsMpu(sizes) || containsPortraitInterstitial(sizes)) {
+					return 3471436;
+				}
+				if (containsMobileSticky(sizes)) {
+					return 3471440;
+				}
+				break;
+			default:
+				return -1;
+		}
+	}
+
+	if (isInAuOrNz()) {
+		switch (getBreakpointKey()) {
+			case 'D':
+				if (containsMpuOrDmpu(sizes) || containsWS(sizes)) {
+					return 3471452;
+				}
+				// top-above-nav on desktop
+				if (
+					containsLeaderboardOrBillboard(sizes) &&
+					slotId === 'dfp-ad--top-above-nav'
+				) {
+					return 3471458;
+				}
+				// Fronts-banners on desktop
+				if (
+					containsBillboard(sizes) &&
+					slotId.includes('fronts-banner')
+				) {
+					return 3471462;
+				}
+				break;
+			case 'M':
+				if (containsMpu(sizes) || containsPortraitInterstitial(sizes)) {
+					return 3471464;
+				}
+				if (containsMobileSticky(sizes)) {
+					return 3471468;
+				}
+				break;
+			default:
+				return -1;
+		}
+	}
```

Comment:
This could be written with a bit less code by having the region check inside the size check:
```suggestion
	switch (getBreakpointKey()) {
		case 'D':
			if (containsMpuOrDmpu(sizes) || containsWS(sizes)) {
				if (isInUk()) {
					return ...'
				} else if (isInRow()) {
					...
				} else if (isInUsOrCa()) {
				    ...
				} else if (isInAuOrNz()) {
				 ...
				}
				
			}
			// top-above-nav on desktop
			if (
				containsLeaderboardOrBillboard(sizes) &&
				slotId === 'dfp-ad--top-above-nav'
			) {
				if (isInUk()) {
					return ...'
				} else if (isInRow()) {
					...
				} else if (isInUsOrCa()) {
				    ...
				} else if (isInAuOrNz()) {
				 ...
				}
			}
			// Fronts-banners on desktop
			if (
				containsBillboard(sizes) &&
				slotId.includes('fronts-banner')
			) {
				if (isInUk()) {
					return ...'
				} else if (isInRow()) {
					...
				} else if (isInUsOrCa()) {
				    ...
				} else if (isInAuOrNz()) {
				 ...
				}
			}
			break;
		case 'M':
			if (containsMpu(sizes) || containsPortraitInterstitial(sizes)) {
				if (isInUk()) {
					return ...'
				} else if (isInRow()) {
					...
				} else if (isInUsOrCa()) {
				    ...
				} else if (isInAuOrNz()) {
				 ...
				}
			}
			break;
		default:
			return -1;
	}
	
```